hyperstone/plugins/hooks/fake_dll.py

```diff
@@ -21,15 +21,31 @@ def _prepare(self):
         self._pe = Plugin.require(PELoader, self.emu)
         super()._prepare()
 
-    def _resolve_export_fallback(self, object_name: str, function_name: str) -> int | None:
+    def _resolve_export_fallback(
+        self, object_name: str, function_name: str
+    ) -> int | None:
         if (
-            object_name in self._pe._phantom_hooks
-            and function_name in self._pe._phantom_hooks[object_name]
+            object_name in self._pe._fake_exports
+            and function_name in self._pe._fake_exports[object_name].functions
         ):
-            return self._pe._phantom_hooks[object_name][function_name]
+            return self._pe._fake_exports[object_name].functions[function_name]
+
+        if object_name.lower() in self._pe._loaded:
+            function_address = self._pe._loaded[object_name.lower()].function(
+                function_name
+            )
+            if function_address:
+                return function_address
 
     def _create_or_resolve_object_handle(self, name: str) -> int:
-        return calculate_aslr(self.emu.arch.bits == 64, True)
+        if name.lower() not in self._pe._loaded:
+            self._pe.interact(PELoaderInfo(name))

```

Comment:
Edge case when `self._pe.ready == False`
Since interact() might be async, this could make it so `name` isn't in `_loaded` yet
You need to `self._pe.prepare(self.emu)` after the `require()`